examples/aishell/asr1/READEME.md

```diff
@@ -0,0 +1,301 @@
+
+```bash
+ if [ ${stage} -le 4 ] && [ ${stop_stage} -ge 4 ]; then
+     # ctc alignment of test data
+     CUDA_VISIBLE_DEVICES=0 ./local/align.sh ${conf_path} exp/${ckpt}/checkpoints/${avg_ckpt} || exit -1
+ fi
+```
+
+If you want to train the model, test it and do the alignment,  you can use the script below to execute the stage 0, stage 1,  stage 2, and stage 3 :
+
+```bash
+bash run.sh --stage 0 --stop_stage 4
+```
+
+or if you only need to train a model and do the alignment, you can use these scripts to escape the stage 3(test stage):
+
+```bash
+bash run.sh --stage 0 --stop_stage 2
+bash run.sh --stage 4 --stop_stage 4
+```
+
+or you can also use these scripts in command line (only use CPU).
+
+```bash
+source path.sh
+source ${MAIN_ROOT}/utils/parse_options.sh
+bash ./local/data.sh
+CUDA_VISIBLE_DEVICES= ./local/train.sh conf/conformer.yaml conformer
+avg.sh best exp/conformer/checkpoints 20
+# test stage is optional
+CUDA_VISIBLE_DEVICES= ./local/test.sh conf/conformer.yaml exp/conformer/checkpoints/avg_20
+CUDA_VISIBLE_DEVICES= ./local/align.sh conf/conformer.yaml exp/conformer/checkpoints/avg_20
+```
+
+
+
+## Stage 5: Single audio file inference
+
+In some situation, you want to use the trained model to do the inference for the single audio file. You can use the stage  5. The code is shown below
+
+```bash
+ if [ ${stage} -le 5 ] && [ ${stop_stage} -ge 5 ]; then
+     # test a single .wav file
+     CUDA_VISIBLE_DEVICES=0 ./local/test_hub.sh ${conf_path} exp/${ckpt}/checkpoints/${avg_ckpt} ${audio_file} || exit -1
+ fi
+```
+
+you can train the model by yourself using ```bash run.sh --stage 0 --stop_stage 3```， or you can download the pretrained model by the script below:
+
+```
+wget https://deepspeech.bj.bcebos.com/release2.1/aishell/s1/aishell.release.tar.gz
+tar aishell.release.tar.gz
+```
+
+You need to prepare an audio file, please confirme the sample rate of the audio is 16K.  Assume the path of the audio file is ```data/test_audio.wav```,  you can get the result by runing the script below.
+
+```bash
+CUDA_VISIBLE_DEVICES= ./local/test_hub.sh conf/conformer.yaml exp/conformer/checkpoints/avg_20 data/test_audio.wav
+```
+
+
+
+## Stage 51: Static model transforming（not supported at now）
+
+To transform the dynamic model to static model，stage 51 can be used. The code of this stage is shown below:
+
+```bash
+ if [ ${stage} -le 5 ] && [ ${stop_stage} -ge 5 ]; then
+     # export ckpt avg_n
+     CUDA_VISIBLE_DEVICES=0 ./local/export.sh ${conf_path} exp/${ckpt}/checkpoints/${avg_ckpt} exp/${ckpt}/checkpoints/${avg_ckpt}.jit
+ fi
+```
+
+It is not supported at now, so we set a large stage number for this stage.
+
+
+
+## Stage: 101 Language model training and TLG building (Need further installation! )
+
+You need to install the kaldi and srilm to use the stage 101,  it is used for train language model and build TLG. To do further installation, you need to do these:
```

Comment:
use the stage 101,  it is used for train language model and build TLG. -> use stage 101, it is used for training language model and building TLG.